src/plugins/intel_gpu/src/graph/impls/cpu/detection_output.cpp

```diff
@@ -848,8 +851,15 @@ struct detection_output_impl : typed_primitive_impl<detection_output> {
             generate_detections<ov::element_type_traits<data_types::f16>::value_type>(stream, instance, num_of_images, bboxes, confidences, scoreIndexPairs);
         }
 
-        ev->set();
-        return ev;
+        if (pass_through_events) {
+            if (events.size() > 1) {
+                return stream.group_events(events);
+            } else if (events.size() == 1) {
+                return events[0];
+            }
+        }
+
+        return stream.create_user_event(true);
```

Comment:
Won't there be any chance where 
1. current node is in shape_of_subgraph 
2. dependency is from both shape_of_subgraph + gpu impl, and this node need to wait for gpu impl

Reply:
In my understanding, only cpu_impl can be in shape_of_subgraph. And, if a node has a dependency from gpu_impl, `can_execute_in_subgraph` is set as false at L30 in the below code.
https://github.com/openvinotoolkit/openvino/blob/7a1cd37b9db5b31b36ef1ae646bdedebdfbac261/src/plugins/intel_gpu/src/graph/graph_optimizer/mark_shape_of_subgraphs.cpp#L17-L33
So, if a node has a dependency from gpu_impl, this node is not in shape_of_subgraph.
Please correct me if I misunderstood it.